query/releasesearch.go

```diff
@@ -0,0 +1,215 @@
+package query
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+)
+
+type ParamValidator map[paramName]validator
+
+func (qpv ParamValidator) Validate(name, value string) (interface{}, error) {
```

Comment:
(question): can this return a specific type rather than a generic interface ?

Reply:
not really I think -  the idea is that the validation may produce a value of any type, such as date and `sort in thos case, and the caller knows the type of the parameter and can assert that on the return value.
An alternative of providing an actual specific interface would be not be viable; there is one alternative which is to have a signature of
Validate(name, value string, actualValue interface{}) error
where the caller then provides a pointer to the parameter, like passing a pointer when calling
json.Unmarshal(value []byte, actualValue interface{}) error

Note sure it adds a lot in this instance - what do you think?

Reply:
sure, that's fine. We can leave it as it is I guess.